Telemetry/Telemetry.swift

```diff
@@ -153,4 +155,12 @@ public class Telemetry {
         
         pingBuilder.search(location: location, searchEngine: searchEngine)
     }
+
+    // To modify the final key-value data dict before it gets stored as JSON, install a handler using this func.
+    public func beforeSerializePing(pingType: String, handler: @escaping BeforeSerializePingHandler) {
+        if beforeSerializePingHandlers[pingType] == nil {
+            beforeSerializePingHandlers[pingType] = [BeforeSerializePingHandler]()
+        }
+        beforeSerializePingHandlers[pingType]!.append(handler)
```

Comment:
I despise forced unwraps, but I always end up doing it for dict init like this. I should google for the latest cool-cat-swift way.